bin/maintenance/update_header.py

```diff
@@ -179,31 +181,35 @@ def main():
         sys.exit(1)
 
     project, year, path, file_ = _process_args(sys.argv[1:])
+    error = False
 
     if path is not None:
         print(cformat("Updating headers to the year %{yellow!}{year}%{reset} for all the files in "
                       "%{yellow!}{path}%{reset}...").format(year=year, path=path))
         for root, _, filenames in os.walk(path):
             for filename in filenames:
                 if not blacklisted(path, root):
-                    update_header(project, os.path.join(root, filename), year)
+                    error = update_header(project, os.path.join(root, filename), year) or error
     elif file_ is not None:
         print(cformat("Updating headers to the year %{yellow!}{year}%{reset} for the file "
                       "%{yellow!}{file}%{reset}...").format(year=year, file=file_))
-        update_header(project, file_, year)
+        error = update_header(project, file_, year) or error
```

Comment:
```suggestion
        if update_header(project, file_, year):
            error = True
```

Reply:
I was trying to avoid nesting it further. But that works too!